install.sh

```diff
@@ -0,0 +1,163 @@
+error_handler() {
+    local line_no=$1
+    echo ""
+    echo -e "${RED}${BOLD}✗ Error occurred in install.sh at line ${line_no}${NC}"
+    echo -e "${RED}Installation failed. Please check the error message above.${NC}"
+    echo ""
+    echo -e "${YELLOW}Troubleshooting tips:${NC}"
+    echo "  - Check your internet connection"
+    echo "  - Ensure you have sufficient disk space"
+    echo "  - Try running the failed script individually from scripts/"
+    echo "  - Check the GitHub repository for known issues"
+    echo ""
+    exit 1
+}
+
+# Set up error trap
+trap 'error_handler ${LINENO}' ERR
+
+# Get the directory where this script is located
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+# Determine if we need to clone the repository
+# Check for .git directory - more robust than checking for scripts directory
+if [ ! -d "${SCRIPT_DIR}/.git" ] || [ ! -d "${SCRIPT_DIR}/scripts" ]; then
+    echo -e "${YELLOW}Repository not found locally. Cloning from GitHub...${NC}"
+    echo ""
+    
+    # Verify git is available
+    if ! command -v git &> /dev/null; then
+        echo -e "${RED}✗ Git is not installed${NC}"
+        echo -e "${YELLOW}On macOS, you can install git using:${NC}"
+        echo "  xcode-select --install"
+        echo ""
+        echo "After installing git, run this script again."
+        exit 1
+    fi
+    
+    # Clone into ~/kamaete
+    CLONE_DIR="${HOME}/kamaete"
+    
+    # Check if directory already exists
+    if [ -d "${CLONE_DIR}" ]; then
+        echo -e "${YELLOW}Found existing kamaete directory at ${CLONE_DIR}${NC}"
+        echo -e "${YELLOW}Updating repository...${NC}"
+        cd "${CLONE_DIR}"
+        
+        # Try to update, fail if it doesn't work
+        if ! git pull origin main; then
+            echo ""
+            echo -e "${RED}✗ Failed to update repository${NC}"
+            echo -e "${YELLOW}Please manually update and try again:${NC}"
+            echo "  cd ${CLONE_DIR}"
+            echo "  git pull origin main"
+            exit 1
+        fi
+        
+        echo -e "${GREEN}✓ Repository updated successfully${NC}"
+    else
+        echo -e "${GREEN}Cloning kamaete to ${CLONE_DIR}...${NC}"
+        git clone https://github.com/angelocordon/kamaete.git "${CLONE_DIR}"
+        echo -e "${GREEN}✓ Repository cloned successfully${NC}"
+    fi
+    
+    REPO_ROOT="${CLONE_DIR}"
+    cd "${REPO_ROOT}"
+else
+    REPO_ROOT="${SCRIPT_DIR}"
+fi
+
+# Print welcome banner
+echo ""
+echo -e "${BLUE}${BOLD}========================================${NC}"
+echo -e "${BLUE}${BOLD}   Kamaete (構えて) - Get Ready!${NC}"
+echo -e "${BLUE}${BOLD}========================================${NC}"
+echo ""
+echo -e "${GREEN}This script will set up your macOS development environment${NC}"
+echo -e "${GREEN}All operations are idempotent and safe to run multiple times${NC}"
+echo ""
+echo -e "${YELLOW}Repository location:${NC} ${REPO_ROOT}"
+echo ""
+
+# Verify we're on macOS
```

Comment:
We can safely assume that this will only be run on macOS machines. We can remove this block of code.